bci_tester/data.py

```diff
@@ -347,6 +345,7 @@ def create_BCI(
         build_tag=f"{BCI_CONTAINER_PREFIX}/bci-base:{OS_CONTAINER_TAG}",
         image_type="kiwi",
         bci_type=ImageType.OS,
+        available_versions=[OS_VERSION],
```

Comment:
why not `_DEFAULT_NONBASE_OS_VERSIONS` ?

Reply:
why is this not needed for the tumbleweed case above?

Reply:
Because that's wrong, it includes `15.6` and `tumbleweed`, which is most certainly not the os_version for which `bci-base:15.5` is available.

Reply:
it seems to me the issue is that available_versions for base  (so not nonbase) containers is not expanded properly. 

everything appears to be working with just this change, which would be logical to me:

```patch
--- a/bci_tester/data.py
+++ b/bci_tester/data.py
@@ -53,6 +53,9 @@ _DEFAULT_NONBASE_SLE_VERSIONS = ("15.5", "15.6")
 # Test Language and Application containers by default for these versions
 _DEFAULT_NONBASE_OS_VERSIONS = ("15.6", "tumbleweed")
 
+# Test base containers by default for these versions
+_DEFAULT_BASE_OS_VERSIONS = ("15.5", "15.6", "tumbleweed")
+
 assert (
     sorted(ALLOWED_BASE_OS_VERSIONS) == list(ALLOWED_BASE_OS_VERSIONS)
 ), f"list ALLOWED_BASE_OS_VERSIONS must be sorted, but got {ALLOWED_BASE_OS_VERSIONS}"
@@ -298,6 +301,15 @@ def create_BCI(
                     )
         else:
             available_versions = list(_DEFAULT_NONBASE_OS_VERSIONS)
+    else:
+        if available_versions:
+            for ver in available_versions:
+                if ver not in ALLOWED_BASE_OS_VERSIONS:
+                    raise ValueError(
+                        f"Invalid os version for base container: {ver}"
+                    )
+        else:
+            available_versions = list(_DEFAULT_BASE_OS_VERSIONS)
 
     if available_versions:
```

Reply:
submitted now as alternative here: https://github.com/SUSE/BCI-tests/pull/531